### PR TITLE
Setting up free approach

### PR DESCRIPTION
**Pull request by zaeemansari70**
There were some small points I already added last week when I saw those changes were necessary, I've still added a `#Review` above them so you can review them too.

### CODE REVIEW

File: docs/partial_source/contributing/setting_up.rst [hunk 1]
```diff
@@ -34,6 +34,27 @@ Depending on your preferred mode of cloning, any of the below should work:
 
     gh repo clone --recurse-submodules unifyai/ivy your_folder
 
+A Free Guide To Setting Up
```
Comment (VedPatwardhan) [inline docs/partial_source/contributing/setting_up.rst:37]: I think the header should be "Setting up for Free" and we should mention something along the lines of "Pycharm professional license is only free for students and VSCode is a free alternative". Also the positioning of this section should be below PyCharm given that PyCharm should still be the one that's recommended.

**Review by VedPatwardhan — CHANGES_REQUESTED** [review]
Hi @zaeemansari70, just suggested a few changes, feel free to re-request my review once you're done. Thanks 🙂

File: docs/partial_source/contributing/setting_up.rst (hunk 1 above)
Comment (zaeemansari70) [reply docs/partial_source/contributing/setting_up.rst]: Done!
Do let me know if any other changes are required. Thanks 🙂

File: docs/partial_source/contributing/setting_up.rst [hunk 2]
```diff
@@ -390,11 +395,32 @@ Now, if Hypothesis detects an error in the code it will return more detailed inf
 For questions, please reach out on the `setting up discussion`_
 or on `discord`_ in the `docker channel`_!
 
+Setting up for Free
+-------------------
+
+PyCharm professional is only free for students, a good alternative would be using Visual Studie Code if you don't have a student account. The most easiest and the most efficient way would be using Visual Studio Code with the Docker extension. You'll hopefully be done with this in no time.
```
Comment (VedPatwardhan) [inline docs/partial_source/contributing/setting_up.rst:401]: Could you please frame the first sentence as `Visual Studio Code is a recommended free alternative to setting up, especially if you're not eligible for a student license with PyCharm Professional`?

**Review by VedPatwardhan — CHANGES_REQUESTED** [review]
Hi @zaeemansari70, just one last change and we're good to merge the PR. Thanks 🙂

File: docs/partial_source/contributing/setting_up.rst (hunk 2 above)
Comment (zaeemansari70) [reply docs/partial_source/contributing/setting_up.rst]: Done!

**Pull request merged**